db/migrate/20160113231834_migration_fix_missing_city_id.rb

```diff
@@ -0,0 +1,28 @@
+class MigrationFixMissingCityId < ActiveRecord::Migration
+  def up
+    execute <<-SQL
+-- Old data won't be validated here
+ALTER TABLE projects DISABLE TRIGGER sent_validation;
+
+-- Cases where the city is BH
+UPDATE projects
+SET city_id = 1634
+WHERE
+    city_id IS NULL
+    AND projects.state_order >= 'published'
+    AND EXISTS (SELECT true FROM users u WHERE u.id = projects.user_id AND u.address_city = 'BH');
+
+-- Cases where the city is BH
+UPDATE projects
+SET city_id = 5274
+WHERE
```

Comment:
The SP is the default, so this is a catch-all for everything else with null city_id.
The remaining cases had empty cities, or cities made no sense (aaaaa, asdfg, etc)